### ISSUE TITLE
use of connection: local prevents delgate_to another host

### ISSUE THREAD

**Issue by stevelle** [ansible/ansible#13134]
Ansible Version: 1.9.4
Ansible Configuraiton: n/a
Environment: Ubuntu 14.04
Summary: `connection: local` specified in playbook or through command line prevents `delegate_to` for non-local hosts.
Reproduction: 
- run playbook: `ansible-playbook connection_local_repro.yml -e instance=SOME_HOST -v`
- playbook follows:

```

---
- hosts: 127.0.0.1
  connection: local

  tasks:
  - name: Local Where am i
    command: hostname
    register: localhostname

  - name: Add Instance to inventory
    add_host: name={{ instance }} groups=temp
    tags:
      - instance

  - name: Remote Where am i
    command: hostname
    register: target_hostname
    delegate_to: "{{ instance }}"

  - fail: Bug reproduced
    when: localhostname.stdout == target_hostname.stdout

```

Expected: Remote tasks should indicate remote hostname. This can be demonstrated by commenting out `connection:local` in the above play.

Actual: Remote tasks indicate the local hostname.

```
TASK: [Remote Where am i] ***************************************************** 
<10.10.12.88> REMOTE_MODULE command hostname
<10.10.12.88> EXEC ['/bin/sh', '-c', 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581 && echo $HOME/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581']
<10.10.12.88> PUT /tmp/tmpoYRIlf TO /home/vagrant/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581/command
<10.10.12.88> EXEC ['/bin/sh', '-c', u'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581/command; rm -rf /home/vagrant/.ansible/tmp/ansible-tmp-1447298537.31-54161689028581/ >/dev/null 2>&1']
changed: [127.0.0.1 -> 10.10.12.88] => {"changed": true, "cmd": ["hostname"], "delta": "0:00:00.001533", "end": "2015-11-12 03:22:17.347083", "rc": 0, "start": "2015-11-12 03:22:17.345550", "stderr": "", "stdout": "deployhost", "warnings": []}
```

It is clear that the command is executing through a local connection rather than through ssh.

Workarounds:
- remove connection:local from the playbook and ensure ansible_connection=local is set for localhost in inventory, (e.g. `localhost  ansible_connection=local`)
- restructure to isolate connection:local to a playbook that does not need to use delegate_to 

I am recording this issue primarily to document the workarounds for others.

**Comment by ktosiek** [comment]
Still a problem in 2.0 branch

**Comment by bcoca** [comment]
So i believe #15024 should fix this and apply the correct transport

**Comment by bcoca** [comment]
closing via merged #15024

**Comment by OluOyedipe** [comment]
Hi @bcoca i too am seeing a problem when i set connection to local during a play. i'm using the vsphere_guest module to provision a vm so i've set the connection to local, but when i try to set the hostname of the newly provisioned vm using a delegate_to, the hostname module executes on the local machine. Any ideas what could be the issue?
